imap_processing/codice/codice_l1a_lo_angular.py

```diff
@@ -0,0 +1,359 @@
+"""CoDICE Lo Angular L1A processing functions."""
+
+import logging
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.codice import constants
+from imap_processing.codice.decompress import decompress
+from imap_processing.codice.utils import (
+    CODICEAPID,
+    ViewTabInfo,
+    calculate_acq_time_per_step,
+    get_codice_epoch_time,
+    get_collapse_pattern_shape,
+    get_view_tab_info,
+    index_to_position,
+    read_sci_lut,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _despin_species_data(
+    species_data: np.ndarray, sci_lut_data: dict, view_tab_obj: ViewTabInfo
+) -> np.ndarray:
+    """
+    Apply despinning mapping for angular products.
+
+    Despinned data shape is (num_packets, num_species, 24, inst_az) where
+    we expand spin_sector to 24 by filling with zeros in 12 to 24 or 0 to 11
+    based on pixel orientation.
+
+    Parameters
+    ----------
+    species_data : np.ndarray
+        The species data array to be despun.
+    sci_lut_data : dict
+        The science LUT data used for despinning.
+    view_tab_obj : ViewTabInfo
+        The view table information object.
+
+    Returns
+    -------
+    np.ndarray
+        The despun species data array in
+        (num_packets, num_species, esa_steps, 24, inst_az).
+    """
+    # species_data shape: (num_packets, num_species, esa_steps, *collapsed_dims)
+    num_packets, num_species, esa_steps = species_data.shape[:3]
+    collapsed_dims = species_data.shape[3:]
+    inst_az_dim = collapsed_dims[-1]
+
+    # Prepare despinning output: (num_packets, num_species, esa_steps, 24, inst_az_dim)
+    # 24 is derived by multiplying spin sector dim from collapse table by 2
+    spin_sector_len = collapsed_dims[-2] * 2
+    despun_shape = (num_packets, num_species, esa_steps, spin_sector_len, inst_az_dim)
+    despun_data = np.full(despun_shape, 0)
+
+    # Pixel orientation array and mapping positions
+    pixel_orientation = np.array(
+        sci_lut_data["lo_stepping_tab"].get("pixel_orientation").get("data")
```

Comment:
It might be better to avoid the get() call because then you'd get a nicer explicit KeyError if it doesn't exist.
```suggestion
        sci_lut_data["lo_stepping_tab"]["pixel_orientation"]["data"]
```